x-pack/legacy/plugins/lens/server/usage/visualization_counts.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { KibanaConfig } from 'src/legacy/server/kbn_server';
+import { CallClusterOptions } from 'src/legacy/core_plugins/elasticsearch';
+import { SearchParams, SearchResponse } from 'elasticsearch';
+import { LensUsage } from './types';
+
+type ClusterSearchType = (
+  endpoint: 'search',
+  params: SearchParams & {
+    rest_total_hits_as_int: boolean;
+  },
+  options?: CallClusterOptions
+) => Promise<SearchResponse<unknown>>;
+
+export async function getVisualizationCounts(
+  callCluster: ClusterSearchType,
+  config: KibanaConfig
+): Promise<LensUsage> {
+  const scriptedMetric = {
+    scripted_metric: {
+      // Each cluster collects its own type data in a key-value Map that looks like:
+      // { lnsDatatable: 5, area_stacked: 3 }
+      init_script: 'state.types = [:]',
+      // The map script relies on having flattened keyword mapping for the Lens saved object,
+      // without this kind of mapping we would not be able to access `lens.state` in painless
+      map_script: `
+        String visType = doc['lens.visualizationType'].value;
+        String niceType = visType == 'lnsXY' ? doc['lens.state.visualization.preferredSeriesType'].value : visType;
```

Comment:
This is fine for now, but I really don't like special casing like this. Eventually, we should solve this more generally so that if other visualizations / Lens plugins have sub-types, we still get metrics about that.